src/FakeItEasy/SelfInitializedFakes/SelfInitializationRule.cs

```diff
@@ -55,7 +56,12 @@ public void Apply(IInterceptedFakeObjectCall fakeObjectCall)
             if (this.recorder.IsRecording)
             {
                 this.wrappedRule.Apply(fakeObjectCall);
-                this.recorder.RecordCall(fakeObjectCall.AsReadOnly());
+                var completedFakeObjectCall = new CompletedFakeObjectCall(
+                    fakeObjectCall.FakedObject,
+                    fakeObjectCall.Method,
+                    fakeObjectCall.Arguments.ToArray(),
+                    fakeObjectCall.AsReadOnly().ReturnValue);
+                this.recorder.RecordCall(completedFakeObjectCall);

```

Comment:
The recorder expects a call that contains the post-call arguments (in order to be able to replay it with the appropriate out and ref arguments), but `AsReadOnly()` returns a call that contains the incoming arguments. So I'm explicitly creating a call that has the expected arguments. It's a little ugly, but it's not for long, since SelfInitializedFakes are being removed in 4.0.0.

btw, @blairconrad, your SelfInitializedFakes library will probably be affected by this fix as well. We could extend `ICompletedFakeObjectCall` to include both the incoming and outgoing arguments, if necessary.